### ISSUE TITLE
[Sound Block] Supporting different audio setups

### ISSUE THREAD

**Issue by NullVoxPopuli** [greshake/i3status-rust#20]
```
 $ RUST_BACKTRACE=1 ./i3status-rs -d ../../../i3status-rust.json
{"version": 1, "click_events": true}
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /checkout/src/libcore/option.rs:323
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:355
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:371
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:549
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:511
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:495
   7: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:471
   8: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:69
   9: core::panicking::panic
             at /checkout/src/libcore/panicking.rs:49
  10: <i3status_rs::blocks::sound::Sound as i3status_rs::block::Block>::update
  11: i3status_rs::main
  12: std::panicking::try::do_call
             at /checkout/src/libstd/panicking.rs:454
  13: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:98
  14: std::rt::lang_start
             at /checkout/src/libstd/panicking.rs:433
             at /checkout/src/libstd/panic.rs:361
             at /checkout/src/libstd/rt.rs:57
  15: __libc_start_main
  16: _start

```

I just tried using this for the first time, cause this project looks awesome. :-)

However, I got the above the error.

I'm using this version: 86ff6bc053274ed3f56ffc7b8eb68545e258c118

i3status-rust.json
```
 $ cat i3status-rust.json 
[
  {"block": "temperature", "interval": 10, "collapsed": false},
  {"block": "disk_space", "path": "/", "alias": "/", "type": "available", "unit": "GB", "interval": 20},
  {"block": "memory", "type":"memory", "format_mem":"{Mup}%", "format_swap":"{SUp}%"},
  {"block": "cpu", "interval": 1},
  {"block": "load", "interval": 1, "format": "{1m}"},
  {"block": "sound"},
  {"block": "time", "interval": 60, "format": "%a %d/%m %R"}
]
```

System info:

```
 $ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty
$ uname -a
Linux Orithyia 4.10.0-19-generic #21-Ubuntu SMP Thu Apr 6 17:04:57 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by NullVoxPopuli** [comment]
oh and cargo/rust:

```
 $ rustc --version
rustc 1.17.0 (56124baa9 2017-04-24)
 $ cargo --version
cargo 0.18.0 (fe7b0cdcf 2017-04-24)
```

**Comment by NullVoxPopuli** [comment]
also, I installed the font awesome fonts via:
```
sudo apt install fonts-font-awesome
```

My bar is:

```
bar {
  font pango:DejaVu Sans Mono, Icons 8

  status_command /home/me/.i3/i3status-rust/target/release/i3status-rs /home/me/.i3/i3status-rust.json --icons awesome --theme solarized-dark

  position top
  colors {
    separator #666666
    background #222222
    statusline #dddddd
    focused_workspace #0088CC #0088CC #ffffff
    active_workspace #333333 #333333 #ffffff
    inactive_workspace #333333 #333333 #888888
    urgent_workspace #2f343a #900000 #ffffff
  }
}
```

**Comment by greshake** [comment]
Hey, start by removing the sound block from your config, I'll fix it tomorrow.

**Comment by NullVoxPopuli** [comment]
thanks!

![2017-05-23-212113_876x30_scrot](https://cloud.githubusercontent.com/assets/199018/26382927/c4fa65a0-3ffd-11e7-9373-3f64a6d9ee79.png)

**Comment by greshake** [comment]
What's the output of `amixer get Master` on your system? The sound block currently only supports amixer.

**Comment by NullVoxPopuli** [comment]
```
$ amixer get Master
amixer: Unable to find simple control 'Master',0
```

my regular i3status (which has a work volume indicator) is formatted like this:
```
volume master {
	device = "default"
	mixer = "Master"
	mixer_idx = 0
	format = "♪ %volume "
}

```
not sure if that helps

**Comment by greshake** [comment]
Just `$ amixer` please... Currently the sound block only uses Master, which doesn't seem to work...

**Comment by NullVoxPopuli** [comment]
```
$ amixer
Simple mixer control 'Mic',0
  Capabilities: cvolume cvolume-joined cswitch cswitch-joined
  Capture channels: Mono
  Limits: Capture 0 - 52
  Mono: Capture 44 [85%] [-27.00dB] [on]
```

**Comment by greshake** [comment]
See, I'm not very familiar with the sound systems in Linux. Your main audio output is apparently not using amixer, so you can either switch to a system that enables you to use amixer or add functionality for your setup to the sound block, and hit me with a PR, that way others with your setup may profit. Good luck, I hope for contributions! :)

**Comment by NullVoxPopuli** [comment]
I'm not familiar with linux sound either. seems to be a bit of a mess.

I think I'm using `pulseaudio`

**Comment by greshake** [comment]
I am, too. But there are usually multiple subsystems working together, going from hardware to the highest software level abstractions, provided by pulse.

**Comment by greshake** [comment]
Looking through the sound system part of the Arch wiki, it appears the Alsa system is indeed very ubiquitous. The arch wiki also suggests using amixer or alsamixer to adjust volume etc.. I'd suggest you check your setup, maybe you can fix it easily :) There definitely should be a 'Master' output listed when calling `$ amixer`.

**Comment by NullVoxPopuli** [comment]
Cool, I'll do some googlin' and post my findings here

**Comment by bbgun7** [comment]
I fixed this issue by installing pulseaudio-alsa on manjaro linux, maybe that'll work for others as well

**Comment by moritzheiber** [comment]
@greshake Is there are reason you're not using [alsa-rs]() but instead are calling `amixer` directly? I would need to pass additional parameters to `amixer` on my system (my soundcard is on index 1 instead of 0), I would reckon a proper API would make it easier to add other configuration options?

**Comment by atheriel** [comment]
@moritzheiber I don't believe we were familiar with that project at the time, but it's a nice find. I'd be happy to see some work moving towards using that library instead of shelling out. It might fix some of the issues I've encountered when hotplugging audio devices, too.

**Comment by atheriel** [comment]
I consider this closed by #245.
